src/item_factory.cpp

```diff
@@ -129,9 +129,32 @@ void Item_factory::finalize() {
             }
         }
         // for ammo not specifying loudness (or an explicit zero) derive value from other properties
-        if( obj.ammo && obj.ammo->loudness < 0 ) {
-            obj.ammo->loudness = std::max( std::max( { obj.ammo->damage, obj.ammo->pierce, obj.ammo->range } ) * 3,
-                                           obj.ammo->recoil / 3 );
+        if( obj.ammo ) {
+            if( obj.ammo->loudness < 0 ) {
+                obj.ammo->loudness = std::max( std::max( { obj.ammo->damage, obj.ammo->pierce, obj.ammo->range } ) * 3,
+                                               obj.ammo->recoil / 3 );
+            }
+
+            const auto &mats = obj.materials;
+            if( std::find( mats.begin(), mats.end(), material_id( "hydrocarbons" ) ) == mats.end() &&
+                std::find( mats.begin(), mats.end(), material_id( "oil" ) ) == mats.end() ) {
+                const auto &ammo_effects = obj.ammo->ammo_effects;
+                obj.ammo->cookoff = ammo_effects.count( "INCENDIARY" ) > 0 ||
```

Comment:
`ammo_effects` is a `std::set` and can only ever contain zero or one of an element which this code does not imply. STL is deficient and should really provide an `exists` or `has` method but as it doesn't consider:

`ammo_effects.count( "INCENDIARY" )`